core/src/main/java/de/danoeh/antennapod/core/service/PlayerWidgetJobService.java

```diff
@@ -109,6 +113,14 @@ private void updateViews() {
                 PlaybackService.getPlayerActivityIntent(this),
                 PendingIntent.FLAG_UPDATE_CURRENT);
 
+        for (int appWidgetId : widgetIds) {
+            SharedPreferences prefs = getSharedPreferences(WIDGET_PREFS, Context.MODE_PRIVATE);
+            int backgroundColor = prefs.getInt(WIDGET_COLOR + appWidgetId, 0x262C31);
+            RemoteViews widgetViews = new RemoteViews(getPackageName(), R.layout.player_widget);
+            widgetViews.setInt(R.id.widgetLayout,"setBackgroundColor", backgroundColor);
+            manager.updateAppWidget(appWidgetId, widgetViews);
+        }
```

Comment:
In the bottom of updateViews we already have a loop for the widget IDs. Those can probably be integrated to prevent sending duplicate updates.